code/modules/antagonists/superheroes/buzzon/buzzon_equipment.dm

```diff
@@ -0,0 +1,294 @@
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/melee/beesword/buzzon/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	if(ishuman(hit_atom))
+		var/mob/living/carbon/human/victim = hit_atom
+		if(istype(victim.get_item_by_slot(ITEM_SLOT_OCLOTHING), /obj/item/clothing/suit/hooded/bee_costume/buzzon))
+			var/obj/item/clothing/suit/hooded/bee_costume/buzzon/suit = victim.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+			if(suit.linked_sword == src)
+				victim.put_in_active_hand(src)
+				victim.visible_message("<span class='warning'>[victim] catches [src] out of the air!</span>")
+				return
+	. = ..()
+
+/mob/living/simple_animal/hostile/bee/toxin_type
+	var/toxin_type = /datum/reagent/toxin
+
+/mob/living/simple_animal/hostile/bee/toxin_type/Initialize()
+	. = ..()
+	assign_reagent(GLOB.chemical_reagents_list[toxin_type])
+
+/mob/living/simple_animal/hostile/bee/toxin_type/cryo
+	toxin_type = /datum/reagent/consumable/frostoil
+
+/obj/item/grenade/spawnergrenade/buzzkill/non_toxic
+	spawner_type = /mob/living/simple_animal/hostile/bee
+	deliveryamt = 5
+
+/obj/item/clothing/glasses/hud/security/sunglasses/blue
+	icon_state = "sunhudmed"
+	glass_colour_type = /datum/client_colour/glass_colour/blue
+
+/obj/item/melee/beesword/buzzon/cryo
+	name = "The Cryostinger"
+	desc = "An advanced sword with built-in frost oil injectors."
+	icon_state = "buzzon_sword_cryo"
+	inhand_icon_state = "buzzon_cryo"
+	worn_icon_state = "buzzon_cryo"
+	force = 3 //A bit lower because we have better toxin
+
+	toxin_type = /datum/reagent/consumable/frostoil
+
+//The Bee Hardsuit
+/obj/item/clothing/head/helmet/space/hardsuit/syndi/buzzon
+	name = "bee hardsuit helmet"
+	desc = "A dual-mode advanced helmet equipped with HUD that looks like a bee's head. It is in travel mode."
+	alt_desc = "A dual-mode advanced helmet equipped with HUD that looks like a bee's head. It is in combat mode."
+	icon_state = "hardsuit1-bee"
+	inhand_icon_state = "s_helmet"
+	hardsuit_type = "bee"
+	visor_flags_inv = 0
+	visor_flags = 0
+	on = FALSE
+
+	armor = list(MELEE = 60, BULLET = 60, LASER = 50, ENERGY = 60, BOMB = 55, BIO = 100, RAD = 70, FIRE = 100, ACID = 100, WOUND = 25)
+
+/obj/item/clothing/head/helmet/space/hardsuit/syndi/buzzon/equipped(mob/user, slot, initial)
+	. = ..()
+	if(!ishuman(user))
+		return
+
+	if(slot != ITEM_SLOT_HEAD)
+		return
+
+	var/mob/living/carbon/human/buzzon = user
+
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	hud.add_hud_to(buzzon)
+	ADD_TRAIT(buzzon, TRAIT_DIAGNOSTIC_HUD, CLOTHING_TRAIT)
+	ADD_TRAIT(buzzon, TRAIT_ROBOTIC_FRIEND, CLOTHING_TRAIT)
+
+/obj/item/clothing/head/helmet/space/hardsuit/syndi/buzzon/dropped(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/buzzon = user
+
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	hud.remove_hud_from(buzzon)
+	REMOVE_TRAIT(buzzon, TRAIT_DIAGNOSTIC_HUD, CLOTHING_TRAIT)
```

Comment:
does this remove the diagnostic hud granted by any diaghuds that buzzon might be wearing in his eyewear slot?

Reply:
oh shit